### PR TITLE
fix the rollout of 'pipelines-as-code-secret' in staging

### PR DESCRIPTION
**Pull request by gabemontero**
The 1.11 bump https://github.com/redhat-appstudio/infra-deployments/pull/1985 and its moving of PaC from `pipelines-as-code` namespace to `openshift-pipeilnes` namespace did not account for refs to the old namespace.

Currently the gitops rollout in staging per https://argocd-server-argocd-staging.apps.stonesoups01ue1.2d3m.p1.openshiftapps.com/applications/argocd-staging/pipeline-service-stone-stg-m01?view=tree&resource= is incomplete and currently stuck on

```
one or more objects failed to apply, reason: namespaces "pipelines-as-code" not found. Retrying attempt #115 at 7:04PM.
```

when processing the `pipelines-as-code-secret` defined in infra-deployments

### CODE REVIEW

**Comment by gabemontero** [comment]
/hold

I minimally want to review the impact of the `base` gitops overlay updates with sufficient SMEs before we merge.

I want to confirm they do not impact prod.

File: components/pipeline-service/base/external-secrets/pipelines-as-code/kustomization.yaml
Comment (gabemontero) [inline components/pipeline-service/base/external-secrets/pipelines-as-code/kustomization.yaml:5]: yeah I think I need to temporarily duplicate components/pipeline-service/base/external-secrets/pipelines-as-code in both staging and prod, where I use the old namespace for pre 1.11, and the new namespace for post 1.11

**Comment by gabemontero** [comment]
test still running but hitting RHTAPBUGS-450

`Components not in ready state: OpenShiftPipelinesAsCode: reconcile again and proceed
Some tekton resources are not ready yet:
Components not in ready state: OpenShiftPipelinesAsCode: reconcile again and proceed`

I'm going to push my update for https://github.com/redhat-appstudio/infra-deployments/pull/2094#discussion_r1259923209 and restart the test

**Comment by sbose78** [comment]
Changes look good to me.

**Comment by gabemontero** [comment]
I should get feedback from 

/assign @Michkov

**Comment by gabemontero** [comment]
the e2e is still running, but the gitops rollout with these changes when I tail the logs live is stuck:

`pipeline-service-in-cluster-local OutOfSync Missing`

going to try with a cluster I provision and preview bootstrap

**Comment by gabemontero** [comment]
if need be I'll pivot and manually create a `pipelines-as-code` namespace from infra-deployments and create the secret in both `pipelines-as-code` and `openshift-pipelines` namespaces

**Comment by gabemontero** [comment]
Yeah from my local repro:

```
one or more objects failed to apply, reason: resource mapping not found for name: "pipelines-as-code-secret" namespace: "openshift-pipelines" from "/dev/shm/3792574126": no matches for kind "ExternalSecret" in version "external-
```

I'm going to pivot to leaving the secret in the old namespace, and creating that namespace in infra-deployments.

**Comment by gabemontero** [comment]
I did save my old changes in another branch just in case

**Comment by gabemontero** [comment]
hmmm .... still seeing `Configured pipelines-as-code-secret secret in openshift-pipelines namespace` in the e2e test log

Officially confused

**Comment by gabemontero** [comment]
and RHTPABUGS-450 again
`Components not in ready state: OpenShiftPipelinesAsCode: reconcile again and proceed`

**Comment by gabemontero** [comment]
> hmmm .... still seeing `Configured pipelines-as-code-secret secret in openshift-pipelines namespace` in the e2e test log as I tail it live
> 
> Officially confused

oh wait, that was from the setup-pac-webhook change that @Michkov helped me with

**Comment by gabemontero** [comment]
/hold cancel

bypassing gitops concern with pivot to set-pac-integration.sh

**Comment by gabemontero** [comment]
/assign @sbose78 

(given his prior review - thanks)

**Comment by gabemontero** [comment]
still hitting https://issues.redhat.com//browse/RHTAPBUGS-450 a lot in CI (I may have a preview.sh work around for that if need be), including the current run as I type, but this is bootstrapping OK for me when I boostrap in preview mode against a personally provisioned 4.12 cluster.

will retest once the test finally times out.

**Comment by gabemontero** [comment]
FYI I've opened https://github.com/redhat-appstudio/infra-deployments/pull/2096 to work around https://issues.redhat.com/browse/SRVKP-3245 and https://issues.redhat.com/browse/RHTAPBUGS-450 .... based on how retests go I may pull that commit in this PR

**Comment by gabemontero** [comment]
> FYI I've opened #2096 to work around https://issues.redhat.com/browse/SRVKP-3245 and https://issues.redhat.com/browse/RHTAPBUGS-450 .... based on how retests go I may pull that commit in this PR

OK hit it again with https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/redhat-appstudio_infra-deployments/2094/pull-ci-redhat-appstudio-infra-deployments-main-appstudio-e2e-tests/1678901580847386624

picking my workaround to this PR, see if that helps

**Comment by gabemontero** [comment]
e2e's passed - that is good wrt to creating the pac secrets in both namespaces being OK, but I just realized that the setup-pac-integration is only called during the bootstrap process

but the staging and prod clusters have already been boostrapped, so this script will not be run against any of those clusters after this PR merges

I think this is proven because I don't see the `pipelines-as-code-secret` secret in the `openshift-pipelines` namespaces in the staging clusters, even after https://github.com/redhat-appstudio/infra-deployments/pull/1985 merged.

and with the move to 1.11 and installing PaC via the operator, with the `pipelines-as-code` namespace being removed, the secret originally created by setup-pac-integration has been removed.

So I think that means we
1) someone manually runs the version of the setup pac integration script in this PR against the 2 staging cluster to create the pac integration secret in both namespaces, as well as re-create the `pipelines-as-code` namespace
2) after we stabilize things, we can see about switching all the other external secret refs to the secret in the old namespace, and then remove that namespace, in subsequent PRs.

Does that sound right @Michkov @sbose78 to you ?

If so, it becomes a question of whether we merge some form of this PR first, or just have someone with sufficient permissions log onto the 2 staging clusters and run the updated script from this PR.

Off to bed - will look for responses in the AM.  Thanks.

File: hack/build/setup-pac-integration.sh
Comment (Michkov) [inline hack/build/setup-pac-integration.sh:87]: This is not problem, staging is not using setup-pac-integration.sh for setup, it's using external secrets, so you need to update deployment of external secret for stage to be into right namespace -> https://github.com/redhat-appstudio/infra-deployments/blob/main/components/pipeline-service/base/external-secrets/pipelines-as-code/kustomization.yaml#L5

**Review by Michkov — CHANGES_REQUESTED** [review]

File: hack/build/setup-pac-integration.sh
Comment (gabemontero) [reply hack/build/setup-pac-integration.sh]: update pushed, along with the configmap patch needing to be updated I believe @Michkov PTAL

File: components/pipeline-service/base/external-secrets/pipelines-as-code/kustomization.yaml
Comment (Michkov) [inline components/pipeline-service/base/external-secrets/pipelines-as-code/kustomization.yaml:5]: this is changing production as well.

Comment (gabemontero) [reply components/pipeline-service/base/external-secrets/pipelines-as-code/kustomization.yaml:5]: right re https://github.com/redhat-appstudio/infra-deployments/pull/2094#discussion_r1259923209 ... discussing in slack

**Review by Michkov — APPROVED** [review]

**Pull request merged**
